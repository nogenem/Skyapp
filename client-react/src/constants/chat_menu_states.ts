export const MENU_STATES = {
  MAIN: 1,
  CHANGING_LANGUAGE: 2,
  CHANGING_USER_STATUS: 3,
  CHANGING_USER_THOUGHTS: 4,
} as const;
type TMenuStates = typeof MENU_STATES[keyof typeof MENU_STATES];

export type { TMenuStates };
