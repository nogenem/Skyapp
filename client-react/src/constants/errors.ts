export const INVALID_EMAIL = "Invalid email";
export const CANT_BE_BLANK = "Can't be blank";
export const PASSWORDS_MUST_MATCH = "Passwords must match";
export const fieldIsTooShort = (n: number) =>
  `This field must be at least ${n} chars long`;
