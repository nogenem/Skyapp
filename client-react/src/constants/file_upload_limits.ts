export default {
  fileSize: 5 * 1024 * 1024, // 5MB in bytes
  files: 5,
};
