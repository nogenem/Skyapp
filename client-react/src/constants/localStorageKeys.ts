export const LOCAL_STORAGE_TOKEN = 'skyapp-jwt';
export const LOCAL_STORAGE_THEME_MODE = 'skyapp-theme-mode';
