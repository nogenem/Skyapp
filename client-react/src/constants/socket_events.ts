export const SOCKET_CONNECT = 'connect';
export const SOCKET_DISCONNECT = 'disconnect';

export const IO_SIGNIN = 'AUTH:SIGNIN';
export const IO_SIGNOUT = 'AUTH:SIGNOUT';
export const IO_NEW_USER = 'AUTH:NEW_USER';

export const IO_GET_INITIAL_DATA = 'CHAT:GET_INITIAL_DATA';
export const IO_PRIVATE_CHANNEL_CREATED = 'CHAT:PRIVATE_CHANNEL_CREATED';
export const IO_GROUP_CHANNEL_CREATED = 'CHAT:GROUP_CHANNEL_CREATED';
export const IO_REMOVED_FROM_GROUP_CHANNEL = 'CHAT:REMOVED_FROM_GROUP_CHANNEL';
export const IO_GROUP_CHANNEL_UPDATED = 'CHAT:GROUP_CHANNEL_UPDATED';
export const IO_MESSAGES_RECEIVED = 'CHAT:MESSAGES_RECEIVED';
export const IO_SET_ACTIVE_CHANNEL = 'CHAT:SET_ACTIVE_CHANNEL';
export const IO_SET_LAST_SEEN = 'CHAT:SET_LAST_SEEN';
export const IO_USER_STATUS_CHANGED = 'CHAT:USER_STATUS_CHANGED';
export const IO_USER_THOUGHTS_CHANGED = 'CHAT:USER_THOUGHTS_CHANGED';
