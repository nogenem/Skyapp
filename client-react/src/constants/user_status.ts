export const USER_STATUS = {
  ACTIVE: 1,
  AWAY: 2,
  DO_NOT_DISTURB: 3,
  INVISIBLE: 4,
} as const;

export const USER_STATUS_2_TEXT = {
  [USER_STATUS.ACTIVE]: 'Active',
  [USER_STATUS.AWAY]: 'Away',
  [USER_STATUS.DO_NOT_DISTURB]: 'Do not disturb',
  [USER_STATUS.INVISIBLE]: 'Invisible',
} as const;

type TUserStatus = typeof USER_STATUS[keyof typeof USER_STATUS];

export type { TUserStatus };
