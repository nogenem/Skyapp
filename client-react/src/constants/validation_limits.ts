export const MIN_PASSWORD_LEN = 6;
