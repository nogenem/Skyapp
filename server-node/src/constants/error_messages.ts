export const INVALID_EMAIL = 'Invalid email';
export const PASSWORDS_MUST_MATCH = 'Passwords must match';
export const fieldIsTooShort = (n: number): string =>
  `This field must be at least ${n} chars long`;
export const INTERNAL_SERVER_ERROR = 'Internal server error';
export const EMAIL_ALREADY_TAKEN = 'This email is already taken';
export const INVALID_ID = 'Invalid id';
export const ROUTE_NOT_FOUND_ERROR = 'Route not found';
export const INVALID_CREDENTIALS = 'Invalid credentials';
export const INVALID_EXPIRED_TOKEN = 'Invalid or expired token';
