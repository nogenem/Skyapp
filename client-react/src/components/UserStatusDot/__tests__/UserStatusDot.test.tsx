import React from 'react';

import { render } from '@testing-library/react';

import { USER_STATUS } from '~/constants/user_status';

import { UserStatusDot } from '../index';

describe('UserStatusDot', () => {
  it('renders all status correctly', () => {
    const { container } = render(
      <UserStatusDot online={false} status={USER_STATUS.ACTIVE} />,
    );
    expect(container).toMatchSnapshot();

    Object.values(USER_STATUS).forEach(status => {
      const { container } = render(<UserStatusDot online status={status} />);
      expect(container).toMatchSnapshot();
    });
  });
});
