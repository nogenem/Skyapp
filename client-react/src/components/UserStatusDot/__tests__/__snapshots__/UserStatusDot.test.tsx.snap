// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStatusDot renders all status correctly 1`] = `
<div>
  <span
    class="makeStyles-dot-1 makeStyles-dot_offline-6 "
    title="Common:Offline"
  />
</div>
`;

exports[`UserStatusDot renders all status correctly 2`] = `
<div>
  <span
    class="makeStyles-dot-1 makeStyles-dot_active-2 "
    title="Common:Active"
  />
</div>
`;

exports[`UserStatusDot renders all status correctly 3`] = `
<div>
  <span
    class="makeStyles-dot-1 makeStyles-dot_away-3 "
    title="Common:Away"
  />
</div>
`;

exports[`UserStatusDot renders all status correctly 4`] = `
<div>
  <span
    class="makeStyles-dot-1 makeStyles-dot_do_not_disturb-4 "
    title="Common:Do not disturb"
  />
</div>
`;

exports[`UserStatusDot renders all status correctly 5`] = `
<div>
  <span
    class="makeStyles-dot-1 makeStyles-dot_invisible-5 "
    title="Common:Invisible"
  />
</div>
`;
